deepchem/data/data_loader.py

```diff
@@ -368,45 +291,91 @@ class CSVLoader(DataLoader):
   pandas, but this class may prove useful if you're processing
   large CSV files that you don't want to manipulate directly in
   memory.
+
+  Examples
+  --------
+  Let's suppose we have some smiles and labels
+
+  >>> smiles = ["C", "CCC"]
+  >>> labels = [1.5, 2.3]
+
+  Let's put these in a dataframe.
+
+  >>> import pandas as pd
+  >>> df = pd.DataFrame(list(zip(smiles, labels)), columns=["smiles", "task1"])
+
+  Let's now write this to disk somewhere. We can now use `CSVLoader` to
+  process this CSV dataset.
+
+  >>> import tempfile
+  >>> import deepchem as dc
+  >>> with tempfile.NamedTemporaryFile(mode='w') as tmpfile:
+  ...   df.to_csv(tmpfile.name)
+  ...   loader = dc.data.CSVLoader(["task1"], feature_field="smiles",
+  ...                              featurizer=dc.feat.CircularFingerprint())
+  ...   dataset = loader.create_dataset(tmpfile.name)
+  >>> len(dataset)
+  2
+
+  Of course in practice you should already have your data in a CSV file if
+  you're using `CSVLoader`. If your data is already in memory, use
+  `InMemoryLoader` instead.
   """
 
   def __init__(self,
-               tasks,
-               smiles_field=None,
+               tasks: OneOrMany[str],
+               feature_field: Optional[str] = None,
+               label_field: Optional[str] = None,
+               weight_field: Optional[str] = None,
+               smiles_field: Optional[str] = None,
                id_field=None,
-               featurizer=None,
+               featurizer: Optional[Featurizer] = None,
                log_every_n=1000):
```

Comment:
Missing type annotations on `id_field` and `log_every_n`.

Reply:
Good catch, will fix!